### DISCUSSION TITLE
Changing text in a URL paramater list

### DISCUSSION THREAD

**Discussion by MD011436** [mganss/HtmlSanitizer#473] [Q&A]
I'm using this sanitizer to prevent Xss. Basically, expecting all parameters to be passed without interpretation. 
My problem is there is a conversion of a parameter that gets changed incorrectly. The issue is with "&timestamp=1687173840"  ends up as    "×tamp=1687173840".     I'm expecting to get "&amp;timestamp=1687173840"

If this does this for "&timestamp="  what other interpretations are happening that I may not want.

Original URL:
"https://inspect.cloud.yoursix.com/?deviceId=1ec8f6d6-f04c-625e-aae5-11d71196d6cb&eventType=loiteringguard&siteId=1ec8f3c0-a82f-6139-81f2-5a4e8eac0955&timestamp=1687173840&token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJjc2kiLCJleHAiOjE2ODc0MzMwNDAsIkNlbnRyYWxTdGF0aW9uSUQiOiIxZWM4ODU5OS0xNzlmLTYyYjctOGRiZi02YmMzODUyZDY0ZTQiLCJTaXRlSUQiOiIxZWM4ZjNjMC1hODJmLTYxMzktODFmMi01YTRlOGVhYzA5NTUiLCJFdmVudElEIjoiMWVlMGU5M2MtOWY1ZS02ZjQ4LWE4NjMtNmIyNjgyYjVhNTY2In0.ap-lCGNfzzheZ4DeJjB4adt1qNrCCDz9Jo8dFC0DEls"

Sanitized URL:
"https://inspect.cloud.yoursix.com/?deviceId=1ec8f6d6-f04c-625e-aae5-11d71196d6cb&amp;eventType=loiteringguard&amp;siteId=1ec8f3c0-a82f-6139-81f2-5a4e8eac0955×tamp=1687173840&amp;token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJjc2kiLCJleHAiOjE2ODc0MzMwNDAsIkNlbnRyYWxTdGF0aW9uSUQiOiIxZWM4ODU5OS0xNzlmLTYyYjctOGRiZi02YmMzODUyZDY0ZTQiLCJTaXRlSUQiOiIxZWM4ZjNjMC1hODJmLTYxMzktODFmMi01YTRlOGVhYzA5NTUiLCJFdmVudElEIjoiMWVlMGU5M2MtOWY1ZS02ZjQ4LWE4NjMtNmIyNjgyYjVhNTY2In0.ap-lCGNfzzheZ4DeJjB4adt1qNrCCDz9Jo8dFC0DEls"

**Comment by MD011436** [comment]
My expectation looks like it got changed, it should say "&amp;timestamp"  not "&timestamp"

**Reply by MD011436** [reply]
the discussion is changing my expectation to the wrong thing. notice the & a m p ;   on the other parameters  but for timestamp it does something different  it gives me × t a m p

**Comment by mganss** [comment]
This is expected behavior. See #362 for more information.

**Comment by mganss** [comment]
See also #190
